tests/sentry/models/test_lastappconnectbuildscheck.py

```diff
@@ -0,0 +1,156 @@
+import uuid
+from datetime import datetime
+
+import pytest
+from django.utils import timezone
+
+from sentry.lang.native import appconnect
+from sentry.models.latestappconnectbuildscheck import LatestAppConnectBuildsCheck
+
+
+class TestCreateOrUpdateBuildCheck:
```

Comment:
cc @wedamija: this is specifically where the `create_and_update` unit tests for `LatestAppConnectBuildCheck` are at

Reply:
After having a look at this I think that rather than checking `create_or_update` directly, it might be a good idea to check that `LatestAppConnectBuildsCheck` is correctly created/updated after `inner_dsym_download` is called.

Reply:
I can look into maybe reworking `inner_dsym_download` so it's possible to unit test the block of logic that utilizes `create_or_update`. I'll admit that I'm balking a little at the idea of mocking up an entire `AppStoreConnect` client _as well_ as an `iTunes` client to test this especially since my python isn't too great, so I'll take a bit of a shortcut: I'll just pull the build processing bits into a helper function which accepts a list of things returned by Apple's API.

With this approach I believe this should alleviate some of the concern that @Swatinem has mentioned around the unit tests not actually testing the business logic.

Reply:
So this doesn't need to be addressed in this PR necessarily, but looking around, are there tests for src/sentry/api/endpoints/project_app_store_connect_credentials.py at all? I don't see any references to `sentry-api-0-project-appstoreconnect-apps`.

cc @RaduW, @flub

Are there tests for this project at all?

Reply:
Pushed changes that test some small part of `inner_dsym_download` instead of just testing `create_and_update` itself. Hopefully that's more along the lines of what you were hoping to see, but let me know if that isn't the case.

> Are there tests for this project at all?

There are a few tests although not necessarily ones that test the endpoints: 
- `tests/sentry/lang/native/test_appconnect.py`
- `tests/sentry/utils/appleconnect/test_itunes_connect.py` (need to be manually run)
- `tests/sentry/tasks/test_app_store_connect.py`

As @Swatinem's mentioned, it's a little difficult to write integration tests for this project in particular. The closest thing we have right now is `test_itunes_connect` which needs to be manually run due to mandatory 2FA.

Reply:
@relaxolotl (or others): Are we able to use `responses` to fake parts of the 2FA check, or otherwise mock it? I'm a bit concerned about having untested code here, in the past this has led to incidents like https://www.notion.so/sentry/2021-05-12-GitHub-Integration-Broken-1217d028c85a4df5a14f633fd7cd014c

Reply:
@wedamija yes, I'm also very concerned about this and this is very likely to come and hurt us.  I finally managed to get the 2FA tests in `test_itnues_connect.py`, but as said they need manual running.  I was thinking of mocking the clients from `src/sentry/utils/appleconnect/` in the other places where this is used like the endpoints and the tasks, which I think at this point has higher impact than using `responses` (but thanks for that tip, I was wondering if we were using anything like this already).  Unfortunately it seems rather hard to convince people currently that we should spend more time on this 😞

Reply:
@flub Ok, if you want me to try to push for some time to work on these tests I can try to do so, just let me know. I'll stop bugging y'all about this now :P

Reply:
I'm all for better test coverage, so if you can apply some influence I'd certainly not complain!
(sorry for delay, so many open tabs...)